TASK-7: Navigation Bar

cristian-aldea opened the pull request:
**Task**

#35

**Pre-requisites**

- [ ] React component hierarchy exists for devs to follow

**Acceptance Criteria**

- [ ] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [ ] Documentation present for every new component and function
```
 * Description of Function/Component
 *
 * @param param1 Description of parameter 1
 * @param param2 Description of parameter 2
```
- [ ] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)


**Notes**
- Removed e2e tests since they were failing and we don't currently have any user story implemented.

leojrsilao committed:
Add navigation setup #35

cristian-aldea committed:
Reorganize project files #35

cristian-aldea committed:
Adjust nav bar styling and format #35

Using <Link> instead of <a>
Using svg as a react component instead of img (easier to apply color, don't need to use css filters)

Also, commented out fetch() done in app.tsx

cristian-aldea committed:
Fixed broken .svg files #35

cristian-aldea committed:
Updated styling and format of NavigationBar #35

- Implemented basic routing in app.tsx
- Updated nav bar styling and add logo
- Use NavLink instead of Link, and set styling dynamically
- Implemented media queries to support mobile/desktop styling

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>
Co-authored-by: Ragith <r_sabapa@live.concordia.ca>

cristian-aldea committed:
Change styling to use flex instead of fixed display #35

Before, the nav bar would hide any content behind it since it was display fixed, now the nav bar is a flex item along with the main content.

Set font to Inter

Clean up styling

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>

cristian-aldea committed:
Add documentation #35

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>

cristian-aldea committed:
Implement testing for nav bar component #35

Add types for react-test-renderer

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>

cristian-aldea committed:
Pass testing, linting and e2e #35

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>

cristian-aldea committed:
Make middle button slightly bigger #35

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>

cristian-aldea committed:
Update snapshots #35

JimLau49 reviewed (dismissed):
Nice work with the media query, sonar cloud needs to be fixed as well so that it can merge but other than that looking good !

Pre-requisites
✔ React component hierarchy exists for devs to follow

Acceptance Criteria

✔ Nav bar shows all the buttons and each button is routed (but can lead to dead links)
✔ Documentation present for every new component and function
✔ Unit and UI tests are implemented for every component and function
✔ Must match Mockup

cristian-aldea committed:
Fix for sonar cloud coverage #35

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>
Co-authored-by: Cindy <lo_cindy@outlook.com>

cristian-aldea committed:
Exclude test files from coverage #35

Co-authored-by: Leo <leojrsilao@gmail.com>
Co-authored-by: Jonathan <bit172@gmail.com>
Co-authored-by: Cindy <lo_cindy@outlook.com>

RGPosadas commented on apps/mull-ui/src/constants.ts:
```diff
@@ -0,0 +1,10 @@
+export const ROUTES = {
+  HOME: '/home',
+  MAP: '/map',
+  NEW_EVENT: '/new-event',
```
Not blocking, but to match with how we named our EPICs/USs/requirements, I would change this name to `/create-event`

cristian-aldea replied:
Sure thing

RGPosadas reviewed (changes requested):
**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

**Blocking Comments**
1. Nice job so far! My only concern is that the icons are bigger compared to their mobile counterpart (desktop is perfectly fine though).
![image](https://user-images.githubusercontent.com/33991121/95404064-084fa300-08e2-11eb-9beb-13dd0793689a.png)

1. The icons are too close to the navbar border on mobile as well.
![image](https://user-images.githubusercontent.com/33991121/95404366-cf63fe00-08e2-11eb-8eb1-30421dacde4f.png)

cristian-aldea committed:
Adjust mobile button styling #35

Co-authored-by: Leo <leojrsilao@gmail.com>

r-saba reviewed (changes requested):
Looking good, just need to support older browsers.

**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

How it looks on my 2019 browser
![image](https://user-images.githubusercontent.com/25574985/95406618-ce35cf80-08e8-11eb-8946-3e97ca1af436.png)

cristian-aldea committed:
Improve styling and compatibility with older browsers #35

the gap property is relatively new, and not support on some older versions of browsers. It was removed in favour of using margins.

Co-authored-by: Leo <leojrsilao@gmail.com>

TheGreatMarkus commented:
@RGPosadas @r-saba thanks for the nice catches. Feel free to let me know of other issues you find

cristian-aldea committed:
Update test id to match new route name #35

RGPosadas reviewed (dismissed):
LGTM! 

**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [x] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

ghiobi commented on apps/mull-ui/src/constants.ts:
```diff
@@ -0,0 +1,10 @@
+/**
+ * Map of the current routes for the application
+ */
+export const ROUTES = {
```
If they are constants and the values in the properties don't change, might as well use enums from TypeScript.

cristian-aldea replied:
Sure why not

cristian-aldea committed:
Fix missing box shadow on mobile #35

cristian-aldea committed:
Change ROUTES from map to enum #35

TheGreatMarkus commented:
@r-saba  @ghiobi I've addressed the issue you brought up, but please don't re-review just yet, I want to have a discussion about the icons in today's scrum.

cristian-aldea committed:
Adjust Nav bar icons #35

Co-authored-by: Leo <leojrsilao@gmail.com>

r-saba reviewed (approved):
Fantastic work! LGTM! I am accepting even though they don't match the current mockups*.

Note* It does not match mockup, but matches what the mockups will look like once they are updated with the newly agreed upon navigation buttons.

**Acceptance Criteria**

- [x] Nav bar shows all the buttons and each button is routed (but can lead to dead links)
- [x] Documentation present for every new component and function
- [x] Unit and UI tests are implemented for every component and function
- [ ] Must match [Mockup](https://www.figma.com/file/FJPXMYI4ULCnuCEZvPs67k/?node-id=534%3A6528)

RGPosadas reviewed (approved):
Wow, buttons look so much sleeker!
My LTGM still stands.

JimLau49 reviewed (approved):
Previous comments still apply, good work 🚀
